skimage/metrics/overlap.py

```diff
@@ -0,0 +1,221 @@
+"""
+Module defining a number of functions to quantify the overlap between shapes.
+for instance rectangles representing detections by bounding-boxes.
+
+"""
+
+class Rectangle():
+    """
+    Represents rectangles which position corresponds to the (r,c) coordinates for the top left corner.
+    Rectangle objects have a number of attributes:
+        - coordinates for the bottom right corner
+        - rectangle dimensions
+        - area
+    as well as utility functions :
+        - computation of the intersection of 2 rectangles
+        - equivalent area for te union of 2 rectangles.
+    """
+
+    def __init__(self, top_left, bottom_right=None, size=None):
+        """
+        Construct a rectangle using the (r,c) coordinates for the top left corner,
+        and either the coordinates of the botton right corner
+        or the rectangle size (height, width).
```

Comment:
Oh I see that you used NumPy here. 🤦 Sorry about that. Actually a weird funkiness of the whole thing is that the *init* docstring goes on the class, and the init typically doesn't have one. That's because `__init__` is a private method and typically does not get rendered in the documentation. It's a bit inconsistent but that's how it is. 🤷

Reply:
I was actually wondering about that 😅, in the Ipython console both are shown but it makes sense indeed to have everything in the class dosctring. I updated accordingly. 

However I also replaced the `area` and `dimensions` attributes with getter methods and I hid the `top_left` , `bottom_right`, `width` and  `height` attributes by preceding their name with an underscore. Just to prevent users to update these values doing myRectangle.topLeft = (r,c), which would invalidate the rest of the attributes. 

I though of making them properties with getter and setter but changing the position of one corner does not tell us if the users wants to also move the other corner (ie move the rectangle) or actually resize the rectangle (ie keeping the other corner in place).